universal-login-contracts/contracts/ERC1077.sol

```diff
@@ -31,18 +31,19 @@ contract ERC1077 is KeyHolder, IERC1077 {
         OperationType operationType,
         bytes memory signatures) public view returns (bool)
     {
-        address signer = getSigner(
-            address(this),
-            to,
-            value,
-            data,
-            nonce,
-            gasPrice,
-            gasToken,
-            gasLimit,
-            operationType,
-            signatures);
-        return keyExist(bytes32(uint256(signer)));
+        address[] memory signers = new address[](signatures.length.div(65));
+        for (uint i = 0;i < signatures.length.div(65); i++) {
+            bytes memory sig = new bytes(65);
+            for (uint j = 0;j < 65; j++) {
```

Comment:
This is pretty expensive, should be done using assembly, extracted to separate method and tested.
Example of such a function can be found [here](https://github.com/gnosis/safe-contracts/blob/a97c6fd24f79c0b159ddd25a10a2ebd3ea2ef926/contracts/common/SignatureDecoder.sol#L33) and usage [here](https://github.com/gnosis/safe-contracts/blob/development/contracts/GnosisSafe.sol#L147)

Reply:
Thank you. that helps a lot.

Reply:
Solium doesn't like solidity inline assemby so i turned off "security/no-inline-assembly" in .soliumrc.json